File: app/src/main/java/com/fsoft/vktest/ViewsLayer/MessagesFragment/MessageStatus.java
package com.fsoft.vktest.ViewsLayer.MessagesFragment;

import com.fsoft.vktest.AnswerInfrastructure.Message;

import java.util.Objects;

public class MessageStatus{
    public final int STATUS_RECEIVED = 0;
    public final int STATUS_ANSWERED = 1;
    public final int STATUS_IGNORED = 2;
    public final int STATUS_ERROR = 3;
    private Message message = null;
    private int status = 0;

    public MessageStatus(Message message, int status) {
        this.message = message;
        this.status = status;
    }

    public MessageStatus(Message message) {
        this.message = message;
    }

    public MessageStatus received(){
        status = STATUS_RECEIVED;
        return this;
    }

    public MessageStatus answered(){
        status = STATUS_ANSWERED;
        return this;
    }

    public MessageStatus error(){
        status = STATUS_ERROR;
        return this;
    }

    public MessageStatus ignored(){
        status = STATUS_IGNORED;
        return this;
    }

    public boolean isReceived(){
        return status == STATUS_RECEIVED;
    }

    public boolean isAnswered(){
        return status == STATUS_ANSWERED;
    }

    public boolean isIgnored(){
        return status == STATUS_IGNORED;
    }

    public boolean isError(){
        return status == STATUS_ERROR;
    }


    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        //чтобы можно было использовать contains (message)
        if(o.getClass() == Message.class)
            return o.equals(message);
        if (o == null || getClass() != o.getClass()) return false;
        MessageStatus that = (MessageStatus) o;
        return Objects.equals(message, that.message);
    }

    @Override
    public int hashCode() {
        return Objects.hash(message);
    }

    public void setMessage(Message message) {
        this.message = message;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public Message getMessage() {
        return message;
    }

    public int getStatus() {
        return status;
    }
}

File: app/src/main/java/com/fsoft/vktest/NewViewsLayer/MessagesFragment/MessageStatus.kt
package com.fsoft.vktest.NewViewsLayer.MessagesFragment

import com.fsoft.vktest.AnswerInfrastructure.Message
import java.util.Objects


class MessageStatus {
    val STATUS_RECEIVED: Int = 0
    val STATUS_ANSWERED: Int = 1
    val STATUS_IGNORED: Int = 2
    val STATUS_ERROR: Int = 3
    var message: Message? = null
    var status: Int = 0

    constructor(message: Message?, status: Int) {
        this.message = message
        this.status = status
    }

    constructor(message: Message?) {
        this.message = message
    }

    fun received(): MessageStatus {
        status = STATUS_RECEIVED
        return this
    }

    fun answered(): MessageStatus {
        status = STATUS_ANSWERED
        return this
    }

    fun error(): MessageStatus {
        status = STATUS_ERROR
        return this
    }

    fun ignored(): MessageStatus {
        status = STATUS_IGNORED
        return this
    }

    val isReceived: Boolean
        get() = status == STATUS_RECEIVED

    val isAnswered: Boolean
        get() = status == STATUS_ANSWERED

    val isIgnored: Boolean
        get() = status == STATUS_IGNORED

    val isError: Boolean
        get() = status == STATUS_ERROR


    override fun equals(o: Any?): Boolean {
        if (this === o) return true
        //чтобы можно было использовать contains (message)
        if (o!!.javaClass == Message::class.java) return o == message
        if (javaClass != o.javaClass) return false
        val that = o as MessageStatus
        return message == that.message
    }

    override fun hashCode(): Int {
        return Objects.hash(message)
    }
}